### PR TITLE
Rename MiddlewareStackInterface to MiddlewarePipelineInterface

### PR DESCRIPTION
**Pull request by devanych**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ✔️
| Fixed issues  | -

### CODE REVIEW

File: src/MiddlewareDispatcher.php [hunk 1]
```diff
@@ -12,32 +12,32 @@
 final class MiddlewareDispatcher
 {
     /**
-     * Contains a stack of middleware handler.
+     * Contains a pipeline of middleware handler.
```
Comment (samdark) [inline src/MiddlewareDispatcher.php:15]:
```suggestion
     * Contains a middleware pipeline handler.
```

File: src/MiddlewareDispatcher.php [hunk 2]
```diff
@@ -12,32 +12,32 @@
...
      *
-     * @var MiddlewareStackInterface stack of middleware
+     * @var MiddlewarePipelineInterface The pipeline of middleware.
```
Comment (samdark) [inline src/MiddlewareDispatcher.php:17]:
```suggestion
     * @var MiddlewarePipelineInterface The middleware pipeline.
```

File: src/MiddlewareDispatcher.php [hunk 3]
```diff
@@ -12,32 +12,32 @@
...
      */
-    private MiddlewareStackInterface $stack;
+    private MiddlewarePipelineInterface $pipeline;
 
-    private MiddlewareFactoryInterface $middlewareFactory;
+    private MiddlewareFactoryInterface $factory;
```
Comment (yiiliveext) [inline src/MiddlewareDispatcher.php:21]: Why we need to rename?

**Review by samdark — APPROVED** [review]

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
👍

**Pull request merged**